python/Ganga/GPIDev/Adapters/IGangaFile.py

```diff
@@ -71,6 +109,36 @@ def put(self):
         """
         raise NotImplementedError
 
+    def copyTo(self, targetPath):
+        """
+        Copy a the file to the local storage using the appropriate file-transfer mechanism
+        This will raise an excpetion if targetPath isn't set to something sensible.
+        Args:
+            targetPath (str): Target path where the file is to copied to
+        """
+        if isinstance(targetPath, str) and targetPath:
+            if regex.search(self.namePattern) is None\
+                and os.path.isfile(os.path.join(self.localDir, self.namePattern)):
+
+                if not os.path.isfile(os.path.join(targetPath, self.namePattern)):
+                    shutil.copy(os.path.join(self.localDir, self.namePattern), os.path.join(targetPath, self.namePattern))
+                else:
+                    logger.debug("Already found file: %s" % os.path.join(targetPath, self.namePattern))
+
+            # Again, cannot perform a remote glob here so have to ignore wildcards
+            else:
+                self.internalCopyTo(targetPath)
+        else:
+            raise GangaException("Cannot perform a copyTo with no given targetPath!")
```

Comment:
For neatness it's usually better to error check and bail early to avoid over-nesting. So:
```python
if not isinstance(targetPath, str):  # I doubt this check is needed though
    raise GangaException("targetPath must be a string")
elif not targetPath:
    raise GangaException("Cannot perform a copyTo with no given targetPath!")
# From here we can know that everything is ok
if regex...
```

See https://programmers.stackexchange.com/questions/18454 and https://en.wikibooks.org/wiki/Computer_Programming/Coding_Style/Minimize_nesting#Early_return